### ISSUE TITLE
Assert in JIT/HardwareIntrinsics/Arm/AdvSimd obj->GetGCSafeMethodTable ()->HasFinalizer ()

### ISSUE THREAD

**Issue by echesakovMSFT** [dotnet/runtime#35718]
As seen in https://dev.azure.com/dnceng/public/_build/results?buildId=626279

Asserts with
```
    JIT\HardwareIntrinsics\Arm\AdvSimd\AdvSimd_r\AdvSimd_r.cmd [FAIL]
      
      Assert failure(PID 5288 [0x000014a8], Thread: 2900 [0x0b54]): obj->GetGCSafeMethodTable ()->HasFinalizer ()
      
      CORECLR! ObjHeader::Validate + 0x36 (0x721af754)
      CORECLR! Object::ValidateInner + 0x279 (0x72359b19)
      CORECLR! Object::Validate + 0xF5 (0x72359865)
      CORECLR! OBJECTREF::OBJECTREF + 0x74 (0x723551c0)
      CORECLR! ObjectFromHandle + 0x28 (0x7213f62c)
      CORECLR! MarshalNative::GCHandleInternalGet + 0x6F (0x725d21bf)
      SYSTEM.PRIVATE.CORELIB! <no symbol> + 0x0 (0x71c9f59e)
      <no module>! <no symbol> + 0x0 (0x0950c467)
      <no module>! <no symbol> + 0x0 (0x0950c289)
      <no module>! <no symbol> + 0x0 (0x0950ba96)
          File: F:\workspace\_work\1\s\src\coreclr\src\vm\syncblk.cpp Line: 2065
          Image: C:\h\w\AD3009B0\p\CoreRun.exe
      
```
in **File: src\coreclr\src\vm\syncblk.cpp Line: 2065** on
* [win-arm](https://helix.dot.net/api/2019-06-17/jobs/1534a252-f658-47b4-add1-264275bf1165/workitems/JIT.HardwareIntrinsics/console)
* [win-arm, COMPlus_TieredCompilation=0](https://helix.dot.net/api/2019-06-17/jobs/8b44ffab-4331-4f58-9384-a9494a970d68/workitems/JIT.HardwareIntrinsics/console)
* [linux-arm](https://helix.dot.net/api/2019-06-17/jobs/d6840114-d543-4f73-82df-149bb8a49c71/workitems/JIT.HardwareIntrinsics/console)
* [linux-arm, COMPlus_TieredCompilation=0](https://helix.dot.net/api/2019-06-17/jobs/064e7516-3b81-4d95-bc80-6535eb2de1b6/workitems/JIT.HardwareIntrinsics/console)
* [win-x86](https://helix.dot.net/api/2019-06-17/jobs/a0e225c8-9e1f-4b88-881c-a73bab3b7dc5/workitems/JIT.HardwareIntrinsics/console)
* [win-x86, COMPlus_TieredCompilation=0](https://helix.dot.net/api/2019-06-17/jobs/d2c45027-c98e-4e2b-95d2-d455c7df3c7c/workitems/JIT.HardwareIntrinsics/console)

@dotnet/jit-contrib

**Comment by echesakovMSFT** [comment]
I am trying to track down the root cause

**Comment by BruceForstall** [comment]
@echesakovMSFT That's from your PR triggered run, right? This morning's outerloop batched job passed: https://dev.azure.com/dnceng/public/_build/results?buildId=626682&view=results

**Comment by echesakovMSFT** [comment]
@BruceForstall We have an ongoing issue with not running coreclr test in CI (see my email from last night) which is being addressed by Steve in #35699

**Comment by echesakovMSFT** [comment]
Identified an issue with the tests
